apps/ecosystem/src/ecosystem.service.ts

```diff
@@ -253,25 +258,169 @@ export class EcosystemService {
     return isEmailSent;
   }
 
+  /**
+   * 
+   * @param RequestSchemaEndorsement 
+   * @returns 
+   */
+  async requestSchemaEndorsement(requestSchemaPayload: RequestSchemaEndorsement, orgId: number): Promise<object> {
+    try {
+      const agentDetails = await this.ecosystemRepository.getAgentDetails(orgId);
+      // eslint-disable-next-line camelcase
+      const platformConfig: platform_config = await this.ecosystemRepository.getPlatformConfigDetails();
+
+      const url = await this.getAgentUrl(agentDetails?.orgAgentTypeId, agentDetails.agentEndPoint);
+
+      const attributeArray = requestSchemaPayload.attributes.map(item => item.attributeName);
+
+      const getEcosystemLeadDetails = await this.ecosystemRepository.getEcosystemLeadDetails();
+
+      const ecosystemLeadAgentDetails = await this.ecosystemRepository.getAgentDetails(Number(getEcosystemLeadDetails.orgId));
+
+      const getEcosystemOrgDetailsByOrgId = await this.ecosystemRepository.getEcosystemOrgDetailsbyId(String(orgId));
+
+      const schemaTransactionPayload: SchemaTransactionPayload = {
+        endorserDid: ecosystemLeadAgentDetails.orgDid,
+        endorse: requestSchemaPayload.endorse,
+        attributes: attributeArray,
+        version: String(requestSchemaPayload.version),
+        name: requestSchemaPayload.name,
+        issuerId: agentDetails.orgDid
+      };
+
+      const schemaTransactionRequest: SchemaMessage = await this._requestSchemaEndorsement(schemaTransactionPayload, url, platformConfig?.sgApiKey);
+      const schemaTransactionResponse: SchemaTransactionResponse = {
+        endorserDid: ecosystemLeadAgentDetails.orgDid,
+        authorDid: agentDetails.orgDid,
+        requestPayload: schemaTransactionRequest.message.schemaState.schemaRequest,
+        status: "Requested",
```

Comment:
Please use the status from `enum`

Reply:
I changed that, it will be reflected in the upcoming PR